### PR TITLE
stubby: init at 0.2.1

### PR DESCRIPTION
**Pull request by leenaars**
###### Motivation for this change

Stubby is a local stub resolver that implements RFC 7858, aka DNS-over-TLS. This allow private DNS requests.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by davidak** [comment]
Can't be built on macOS because of https://github.com/NixOS/nixpkgs/issues/34989.

File: pkgs/tools/networking/stubby/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  meta = with stdenv.lib; {
+    description = "A local DNS Privacy stub resolver (using DNS-over-TLS)";
+    longDescription = ''
+Stubby is an application that acts as a local DNS Privacy stub
+resolver (using RFC 7858, aka DNS-over-TLS). Stubby encrypts DNS
+queries sent from a client machine (desktop or laptop) to a DNS
+Privacy resolver increasing end user privacy. Stubby is developed by
+the getdns team.
+'';
```
Comment (vcunat) [inline pkgs/tools/networking/stubby/default.nix]: Indentation, please, just as it was for getdns in the end https://github.com/NixOS/nixpkgs/pull/34921/files#diff-3f19f5b14f3dc0b7cdf83623bc28eb3cR25

**Comment by Mic92** [comment]
Where is stubby and why is 6ec8fe0 pointing to my commit instead?

**Comment by shlevy** [comment]
I didnt merge this...

**Comment by Mic92** [comment]
w00t?

**Comment by Mic92** [comment]
@leenaars please tell me, you have closed this pull request at least.

**Comment by grahamc** [comment]
```I took a look at the pull request, and I noticed this right above the merge commit:

leenaars force-pushed the leenaars:stubby branch from 282b2b5 to 6ec8fe0 13 days ago

I think this might have triggered an automatic merging of the pull request, which is how people merge pull requests using the command line.

The logic to automatically merge a Pull Request looks at the commit SHA's present on the Pull Request.

If all of the commit SHA's have been pushed to the base branch already, we'll automatically mark it as merged.

Would it be the case that commits that are already present in the base branch NixOS:master were force pushed to the head branch leenaars:stubby, therefore producing what we see now?```

**Comment by leenaars** [comment]
@grahamc: that is indeed what happened. Quite confusing. I've made a new pull request with the (meanwhile updated) stubby. See: https://github.com/NixOS/nixpkgs/pull/36225

**Pull request merged**